cypress/e2e/EngageSphere.cy.js

```diff
@@ -32,64 +32,268 @@ describe("EngageSphere", () => {
+  it("It accepts the cookies banner on first visit", () => {
     // Arrange
-    cy.clearCookie("cookieConsent");
-    cy.reload();
     cy.get('[class^="CookieConsent_banner"]').should("be.visible");
 
     // Act
     cy.contains("button", "Accept").click();
 
     // Assert
     cy.get('[class^="CookieConsent_banner"]').should("not.exist");
+    cy.getCookie("cookieConsent").should("have.property", "value", "accepted");
   });
 
-  it("It renders the 'Hi there' greeting when no name is provided", () => {
+  it("It declines the cookies banner on first visit", () => {
+    // Arrange
+    cy.get('[class^="CookieConsent_banner"]').should("be.visible");
+
+    // Act
+    cy.contains("button", "Decline").click();
+
     // Assert
-    cy.get('[data-testid="name"]').should("be.empty");
-    cy.contains("h2", "Hi there").should("be.visible");
+    cy.get('[class^="CookieConsent_banner"]').should("not.exist");
+    cy.getCookie("cookieConsent").should("have.property", "value", "declined");
   });
+});
 
-  it("It renders the 'Hi Joe' greeting when name is provided", () => {
+describe("Messenger in Mobile Viewport", () => {
+  beforeEach(() => {
+    cy.setCookie("cookieConsent", "accepted");
+    cy.visit("/");
+    cy.viewport("iphone-x");
+  });
+
+  it("It shows the Company name and Action columns and hides the ID, Industry, Number of Employees, and Size columns in a mobile viewport", () => {
+    // Arrange
     // Act
-    cy.get('[data-testid="name"]').type("Joe");
 
     // Assert
-    cy.contains("h2", "Hi Joe").should("be.visible");
+    cy.contains("th", "Company name").should("be.visible");
+    cy.contains("th", "Action").should("be.visible");
+
+    cy.contains("th", "ID").should("not.be.visible");
+    cy.contains("th", "Industry").should("not.be.visible");
+    cy.contains("th", "Number of employees ").should("not.be.visible");
+    cy.contains("th", "Size ").should("not.be.visible");
+  });
+});
+
+describe("EngageSphere API test", () => {
+  beforeEach(() => {});
+
+  it("It returns the correct status and body structure", () => {
+    // Act
+    cy.request({
+      method: "GET",
+      url: "http://localhost:3001/customers",
+      qs: {
+        page: 1,
+        limit: 10,
+        size: "Very Large Enterprise",
+        industry: "Technology",
+      },
+    }).then((response) => {
+      // Assert
+      expect(response.status).to.eq(200);
+
+      const customers = response.body.customers;
+
+      expect(customers).to.be.an("array");
+
+      customers.forEach((customer) => {
+        expect(customer).to.be.an("object");
+        expect(customer).to.include.keys("id", "name", "employees", "size", "industry", "address");
+
+        expect(customer.id).to.be.a("number");
+        expect(customer.name).to.be.a("string");
+        expect(customer.employees).to.be.a("number");
+        expect(customer.size).to.be.a("string");
+        expect(customer.industry).to.be.a("string");
+
+        if (customer.contactInfo !== null) {
```

Comment:
Uma versão simplificada seria a seguinte?

```suggestion
        if (customer.contactInfo) {
```